### PR TITLE
[GIZ-98] Add login/register via id.gov.ua

### PR DESCRIPTION
**Pull request by UvgenGen**
[GIZ-98](https://youtrack.raccoongang.com/issue/GIZ-98) Add login/register via id.gov.ua
- updated requirements
- changed pipeline

### CODE REVIEW

**Review by arsentur — APPROVED** [review]

File: edx_oauth_client/pipeline.py [hunk 1]
```diff
@@ -1,139 +1,164 @@
...
         else:
             user_data = kwargs['response']
+        log.info('Get user data')
 
-        log.info('Get user data: %s', str(user_data))
         data['access_token'] = kwargs['response']['access_token']
 
+        country = user_data.get('country', 'UA')
+
+        if not country:
```
Comment (Golub-Sergey) [inline edx_oauth_client/pipeline.py]: do we need this condition, if we assigned default value for `country` variable in two lines above

File: edx_oauth_client/pipeline.py [hunk 2]
```diff
@@ -1,139 +1,164 @@
...
+            log.info('No country in response.')
+
         for key, value in backend.setting('USER_DATA_KEY_VALUES').items():
             data[key] = user_data.get(value)
 
-        if 'username' not in data or not data['username']:
-            data['username'] = data.get('name', slugify(data['email']))
+        if not data['email']:
```
Comment (Golub-Sergey) [inline edx_oauth_client/pipeline.py:43]: we get `KeyError` if `email` key is not exists in the data

File: edx_oauth_client/pipeline.py [hunk 3]
```diff
@@ -1,139 +1,164 @@
...
+            data['email'] = strategy.session_get('email', "")
 
-        if any((data['first_name'], data['last_name'])):
-            data['name'] = u'{} {}'.format(data['first_name'], data['last_name']).strip()
-        else:
-            data['name'] = user_data.get('preferred_username')
+        data['username'] = data['email']
 
-        if not all((data['username'], data['email'])):
+        if kwargs.get('is_new') and not all((data['username'], data['email'])):
```
Comment (Golub-Sergey) [inline edx_oauth_client/pipeline.py:48]: Do we need to check `not all((data['username'], data['email']))`? We defined them previously

File: edx_oauth_client/pipeline.py [hunk 4]
```diff
@@ -1,139 +1,164 @@
...
-        request = strategy.request
-        data.update({
-            'terms_of_service': 'True',
-            'honor_code': 'True',
-            'password': make_random_password()
-        })
-
-        if request.session.get('ExternalAuthMap'):
-            del request.session['ExternalAuthMap']
+    if not user:
```
Comment (Golub-Sergey) [inline edx_oauth_client/pipeline.py:60]: Do we need to pass this parameters to 'data' and then to the `AccountCreationForm` if user is not exist?

File: edx_oauth_client/pipeline.py (hunk 2 above)
Comment (Golub-Sergey) [reply edx_oauth_client/pipeline.py:43]: Resolved

File: edx_oauth_client/pipeline.py (hunk 4 above)
Comment (Golub-Sergey) [reply edx_oauth_client/pipeline.py:60]: Resolved

**Review by Golub-Sergey — CHANGES_REQUESTED** [review]

File: edx_oauth_client/pipeline.py (hunk 1 above)
Comment (arsentur) [reply edx_oauth_client/pipeline.py]: Fixed

File: edx_oauth_client/pipeline.py (hunk 3 above)
Comment (arsentur) [reply edx_oauth_client/pipeline.py:48]: I think would be better to check here and rise exception if we didn't receive necessary info.

**Comment by Golub-Sergey** [comment]
Good job!)

**Pull request merged**